engine/engine_test.go

```diff
@@ -245,6 +245,15 @@ func TestQueriesAgainstOldEngine(t *testing.T) {
 			    http_requests_total{pod="nginx-2", route="/"} -108+173.00x40`,
 			query: `predict_linear(http_requests_total{route="/"}[1h:1m] offset 1m, 60)`,
 		},
+		{
+			name: "predict_linear with step invariant",
+			load: `load 30s
+			    http_requests_total{pod="nginx-1"} 1+1.1x10
+			    http_requests_total{pod="nginx-2"} 2+2.3x50`,
+			query: `predict_linear({__name__="http_requests_total",pod!~"nginx-1"}[5m] @ start(), -0.37690610678629094)`,
+			end:   time.Unix(600, 0),
+			start: time.Unix(300, 0),
+		},
```

Comment:
Is this an issue for predict linear only or it can impact any function that takes matrix selector with step invariant?

Reply:
It will only impact functions that are at modifier unsafe that takes matrix selector as arg.

https://github.com/prometheus/prometheus/blob/308c8c48c15c74a929c430447df3d3c3a3d4001f/promql/functions.go#L1914

So far only applicable to predict_linear.